lib/reservation_manager.rb

```diff
@@ -0,0 +1,111 @@
+require 'date'
+require_relative 'reservation_dates'
+require_relative 'reservation'
+
+module HotelBookings
+  class Reservation_Manager
+    MAX_ROOMS = 20
+    attr_reader :customer_name, :checkin, :checkout
+   
+    def initialize(customer_name:)
+      @customer_name = customer_name
+
+      @current_reservations = {1 => [], 2 => [], 3 => [], 4 => [], 5 => [], 6 => [], 7 => [], 8 => [], 9 => [], 10 => [], 11 => [], 12 => [], 13 => [], 14 => [], 15 => [], 16 => [], 17 => [], 18 => [], 19 => [], 20 => []}
+    end 
+
+    def rooms 
+      return @current_reservations
+    end 
+
+    def reservation_list(date)
+      unavailable_rooms = []
+      key = 1
+      MAX_ROOMS.times do
+        if @current_reservations[key].include?(date) == true 
+          unavailable_rooms.push(key)
+        end 
+        key += 1
+      end
+      unavailable_rooms = unavailable_rooms.uniq
+      return unavailable_rooms
+    end
+  
+    def available_rooms_list(date)
+      available_rooms = []
+      key = 1
+      MAX_ROOMS.times do 
+        if @current_reservations[key].include?(date) == false 
+          available_rooms.push(key)
+        end 
+        key += 1
+      end
+      available_rooms = available_rooms.uniq
+      return available_rooms
+    end
+
+    def reservation_nights(checkin:, checkout:)
+      @dates = Reservation_Dates.new(checkin: checkin, checkout:checkout)
```

Comment:
Instead of using an instance variable `@dates`, it probably will be fine to keep it as a local variable `dates`. This is because currently, `@dates` isn't used outside of this method, and the `ReservationManager` doesn't need to keep and manage `@dates`.